__test__/pages/advancedSearchForm.test.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react"
+import { fireEvent, render, screen, act } from "@testing-library/react"
+import mockRouter from "next-router-mock"
+import userEvent from "@testing-library/user-event"
+
+import AdvancedSearch from "../../pages/search/advanced"
+
+// Mock next router
+jest.mock("next/router", () => jest.requireActual("next-router-mock"))
+
+describe("Advanced Search Form", () => {
+  const submit = () =>
+    fireEvent(
+      screen.getByRole("button", { name: "Submit" }),
+      new MouseEvent("click")
+    )
+  afterEach(async () => {
+    await act(
+      async () =>
+        await userEvent.click(screen.getByRole("button", { name: "Clear" }))
+    )
+  })
+  it("displays alert when no fields are submitted", () => {
+    render(<AdvancedSearch />)
+
+    submit()
+    screen.getByText(
+      "Please enter at least one field to submit an advanced search."
+    )
+  })
+  it("can set keyword, contributor, title, subject", async () => {
+    render(<AdvancedSearch />)
+
+    const [keywordInput, contributorInput, titleInput, subjectInput] =
+      screen.getAllByRole("textbox")
+    await act(async () => {
```

Comment:
The `await` on the per userEvent line is okay, but you shouldn't need this `act(...)` function wrapper.

Reply:
fixed